backend/src/main/java/ca/bc/gov/hlth/mohums/webclient/ExternalApiCaller.java

```diff
@@ -0,0 +1,89 @@
+package ca.bc.gov.hlth.mohums.webclient;
+
+import org.springframework.http.HttpMethod;
+import org.springframework.http.ResponseEntity;
+import org.springframework.util.MultiValueMap;
+import org.springframework.web.reactive.function.client.WebClient;
+
+import java.util.List;
+
+public class ExternalApiCaller {
```

Comment:
Should there be some unit testing for the methods in this class and the others that have been added?

Reply:
My thoughts in no particular order:

* My advice for this is to write unit tests when you find them useful for developing the functionality, and otherwise focus on integration or acceptance tests; that is, calling the actual REST API.

* A test in `MoHUmsIntegrationTests` will cover `ExternalApiCaller`.

* The advantage to focusing on acceptance tests is that they only to be changed if the business requirements change, for example if the contract of the REST API changes. Test cases should not be tightly coupled to the structure of the code.

* If someone were to refactor the application in the future, they might just throw out ExternalApiCallerTest because the structure would change. They wouldn't throw out `MoHUmsIntegrationTests`.

* I do write unit tests, but primarily as a "test driven development" activity. I write them before I write the code as a sort of outline for what I'm trying to do. Sometimes I delete them or change them into acceptance tests if it's appropriate.

Anyhow, I hope some of that is useful. I'm not asking for any changes.